src/components/common/EffectiveDateTime.vue

```diff
@@ -261,7 +266,9 @@ export default class EffectiveDateTime extends Mixins(DateMixin) {
     // wait for form to update itself before checking validity
     await Vue.nextTick()
 
-    if (this.$refs.form.validate()) {
+    const isDateValid = this.$refs.datePickerRef.$refs.form['validate']()
```

Comment:
What do you think about exposing new methods `validate` and `reset` on the DatePicker component instead?

Reply:
It has the same effect. This is way is faster and keep the component leaner.

Reply:
Yes but we lose typing. And it looks hacky.

I'd rather not try to save a little time today and hope to fix it some day... In my experience, we will be just as busy "some day" as now and it may never get fixed.

Reply:
ok. Will do it.

Reply:
@severinbeauvais I've been looking into it but I couldn't find a way to expose the component's methods other than hacky ways to do it. What do you suggest?

Reply:
Is it possible to call `this.$refs.myComponent.myMethod()`?

If not, leave it alone for now. Good luck :)

Reply:
I tested here and checked that if we change the method's visibility to 'public' it lets you to use it as you are referring to:
https://github.com/bcgov/bcrs-shared-components/blob/master/src/components/DatePicker/DatePicker.vue#L87

Then you can call:
`this.$refs.datePickerRef.clearDate()`